### PR TITLE
New Test File

### PR DESCRIPTION
**Pull request by jyotirmoydotdev**
```bash
[⠒] Compiling...
[⠔] Compiling 4 files with 0.8.20
[⠑] Solc 0.8.20 finished in 2.38s
Compiler run successful with warnings:

Running 6 tests for test/foundry/LoreumToken.t.sol:Deployment
[PASS] test_LoreumToken_burn_restriction_account() (gas: 11538)
[PASS] test_LoreumToken_burn_restriction_amount(uint256) (runs: 256, μ: 23778, ~: 27269)
[PASS] test_LoreumToken_burn_state(uint256) (runs: 256, μ: 34579, ~: 34973)
[PASS] test_LoreumToken_init() (gas: 27697)
[PASS] test_LoreumToken_mint_restriction(uint256) (runs: 256, μ: 26298, ~: 26635)
[PASS] test_LoreumToken_mint_state(uint256) (runs: 256, μ: 35313, ~: 35620)
Test result: ok. 6 passed; 0 failed; 0 skipped; finished in 286.37ms

Running 11 tests for test/foundry/LoreTestCycle.t.sol:LoreumTokenTest
[PASS] testFail_MintFail(uint256) (runs: 256, μ: 21454, ~: 21484)
[PASS] test_AliceBurnToken(uint256) (runs: 256, μ: 73788, ~: 73788)
[PASS] test_AliceBurnTokenFail(uint256) (runs: 256, μ: 36321, ~: 36321)
[PASS] test_Basic() (gas: 40237)
[PASS] test_MintPass(uint256) (runs: 256, μ: 55413, ~: 55413)
[PASS] test_OwnerBurnToken(uint256) (runs: 256, μ: 48055, ~: 48055)
[PASS] test_TokenTransferFail(uint256) (runs: 256, μ: 39542, ~: 39542)
[PASS] test_TokenTransferSuccess(uint256) (runs: 256, μ: 67864, ~: 67864)
[PASS] test_TotalSupplyEqualToAllBalance(uint256) (runs: 256, μ: 70494, ~: 70494)
[PASS] test_transferFrom(uint256) (runs: 256, μ: 87041, ~: 87041)
[PASS] test_transferFromWithIncAllow(uint256) (runs: 256, μ: 124675, ~: 124676)
Test result: ok. 11 passed; 0 failed; 0 skipped; finished in 286.42ms
Ran 2 test suites: 17 tests passed, 0 failed, 0 skipped (17 total tests)
```

### CODE REVIEW

File: test/foundry/LoreTestCycle.t.sol [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+    function setUp() public {
+        lore = new LoreumToken(PremintReceiver, PremintAmount, MaxSupply);
+        alice = address(1);
+        bob = address(2);
+    }
+
+    event Transfer(address indexed from, address indexed to, uint256 value);
+    event Approval(address indexed owner, address indexed spender, uint256 value);
+
+    function invarent() public {
```
Comment (xhad) [inline test/foundry/LoreTestCycle.t.sol:24]: Invariant is misspelled.

File: test/foundry/LoreTestCycle.t.sol [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+        assertGe(lore.maxSupply(), lore.totalSupply());
+        uint256 allBalance = lore.balanceOf(lore.owner()) + lore.balanceOf(alice) + lore.balanceOf(bob);
+        assertEq(lore.totalSupply(), allBalance);
+        assertEq(lore.owner(), address(this));
```
Comment (xhad) [inline test/foundry/LoreTestCycle.t.sol:28]: This contract is ownable and the owner is expected to be changed.

**Review by xhad — CHANGES_REQUESTED** [review]
Kindly add tests for changing the owner of the token.

File: test/foundry/LoreTestCycle.t.sol (hunk 1 above)
Comment (jyotirmoydotdev) [reply test/foundry/LoreTestCycle.t.sol:24]: invarent -> invariant

**Comment by jyotirmoydotdev** [comment]
```bash
[⠢] Compiling...
[⠒] Compiling 1 files with 0.8.20
[⠑] Solc 0.8.20 finished in 2.38s
Compiler run successful with warnings:

Running 6 tests for test/foundry/LoreumToken.t.sol:Deployment
[PASS] test_LoreumToken_burn_restriction_account() (gas: 11538)
[PASS] test_LoreumToken_burn_restriction_amount(uint256) (runs: 256, μ: 22995, ~: 27269)
[PASS] test_LoreumToken_burn_state(uint256) (runs: 256, μ: 34513, ~: 34973)
[PASS] test_LoreumToken_init() (gas: 27697)
[PASS] test_LoreumToken_mint_restriction(uint256) (runs: 256, μ: 26233, ~: 26635)
[PASS] test_LoreumToken_mint_state(uint256) (runs: 256, μ: 35226, ~: 35620)
Test result: ok. 6 passed; 0 failed; 0 skipped; finished in 222.07ms

Running 15 tests for test/foundry/LoreTestCycle.t.sol:LoreumTokenTest
[PASS] invariant() (runs: 256, calls: 3840, reverts: 3023)
[PASS] testFail_MintFail(uint256) (runs: 256, μ: 21432, ~: 21462)
[PASS] test_AliceBurnToken(uint256) (runs: 256, μ: 72112, ~: 72112)
[PASS] test_AliceBurnTokenFail(uint256) (runs: 256, μ: 32690, ~: 32690)
[PASS] test_Basic() (gas: 38604)
[PASS] test_MintPass(uint256) (runs: 256, μ: 53737, ~: 53737)
[PASS] test_OwnerBurnToken(uint256) (runs: 256, μ: 46423, ~: 46423)
[PASS] test_TokenTransferFail(uint256) (runs: 256, μ: 37844, ~: 37844)
[PASS] test_TokenTransferSuccess(uint256) (runs: 256, μ: 66231, ~: 66231)
[PASS] test_TotalSupplyEqualToAllBalance(uint256) (runs: 256, μ: 68841, ~: 68841)
[PASS] test_TransferOwnership() (gas: 37162)
[PASS] test_TransferOwnershipFail() (gas: 32125)
[PASS] test_TransferOwnershipToNULL() (gas: 31915)
[PASS] test_transferFrom(uint256) (runs: 256, μ: 85343, ~: 85343)
[PASS] test_transferFromWithIncAllow(uint256) (runs: 256, μ: 123333, ~: 123334)
Test result: ok. 15 passed; 0 failed; 0 skipped; finished in 572.06ms
Ran 2 test suites: 21 tests passed, 0 failed, 0 skipped (21 total tests)
```

**Review by xhad — APPROVED** [review]

**Pull request merged**